content/articles/setting-up-a-private-dns-server-with-raspberry-pi/index.md

```diff
@@ -0,0 +1,132 @@
+For computers to speak with one another over the web, all participants within the network must have an IP address: Through these addresses, computers recognize precisely which servers/devices they are communicating with. However, it can be difficult to remember IP addresses and hence the need for DNS servers. Domain Name Systems (DNS) changes domain names into IP addresses. A DNS is like a phone book of web addresses that makes access to pages faster. 
+
+### Table of contents
+- [Prerequisites](#prerequisites)
+- [How DNS servers work](#how-dns-servers-work)
+- [Functions of a DNS](#functions-of-a-dns)
+- [Setting up a DNS Server on Raspberry PI](#setting-up-a-dns-server-on-raspberry-pi)
+- [Conclusion](#conclusion)
+
+### Prerequisites
+
+1.	Raspberry Pi (2-4) with Raspbian OS
+2.	Access to the internet
+3.	Access to the Raspberry's terminal
+
+### How DNS servers work
+
+Devices question one or more DNS servers to get access to the correct domain name – IP address translation, a process that can be time-consuming. Therefore, it would be beneficial to shorten the time spent on this process by having a private and dedicated DNS server. A Raspberry Pi, which is small but can be used for various purposes, would be beneficial in such a scenario.
+
+### Functions of a DNS
+
+Below are three significant benefits that a private Domain Name System would offer:
+**Speed** – a query to the web may sometimes have to pass through various servers and routers before the user can access a web page. This time is mostly in milliseconds. However, when a private DNS is set up, this query does not have to pass through the various server, and hence the delivery time will be shorter and more efficient.
+- *Privacy* – to gain access to the correct domain name, information from a device has to be sent to an external server which in turn creates a trail over the internet as opposed to when a private DNS is used, and information does not have to leave the internal network.
+- *Security* – hosting a private DNS server offers increased privacy because the owner gets complete control over entries. This, in turn, protects the server from malicious entries that hackers could facilitate. 
+
+### Setting up a DNS Server on Raspberry PI
+
+#### Step 1:Updating Raspberry Packages 
+
+The first step of the process is updating packages using the following commands:
+
+```sudo apt update```
+
+```sudo apt upgrade```

```

Comment:
This formatting is wrong. 

This is the correct format if you're using the terminal:

```bash
sudo apt update
```
If you're coding in python, it should be:

```python
import numpy as np
```

That's the correct format. Edit all your commands accordingly.